packages/code-du-travail-frontend/src/outils/IndemniteLicenciement/steps/Resultat/store/service.ts

```diff
@@ -29,7 +37,12 @@ export const getInfoWarning = ({
   } else {
     title =
       "Attention il peut quand même exister une indemnité pour le salarié";
-    if (isCdd || agreement?.num === 3239) {
+    if (
+      isCdd ||
+      (contratTravailEligibility &&
+        !informationEligibility &&
+        agreement?.num === 3239)
```

Comment:
C'est un peu dommage de mettre de la logique lié à de la CC là dedans, car le jour où va corriger un bug ou changer la CC, on galèrera à retrouver la ligne

Il faudrait mieux le mettre dans un fichier à part quitte que ça rajoute du code en plus

Reply:
Oui entièrement d'accord. Ca c'est la version quick and dirty vu qu'on veut déployer en mi janvier. Je compte passer ca dans le package modele avec une refacto

Reply:
J'aime bien l'idée discutée ce matin de l'approche via API. Pour cette PR, on pet laisser comme ça je pense, il y a une stratégie à discuter sur l'approche à adopter ici.

Reply:
Oui, mais en attendant autant le mettre dans le folder `agreement`.

Après avoir réfléchi, je trouve pas ça déconnant de mettre la validation côté frontend. Car les messages d'erreurs ne doivent pas faire parti de la partie modèles.

En revanche, la donnée, et l'unique source de vérité, ça serait intéressant de la passer sur publicodes. Le but c'est d'avoir une source de donnée métier.

En l'occurance ici, ce que je passerai côté métier ce sont les `requirements` et non les messages.

Donc oui, mettre en place, une sorte de requirements côté modeles, ça serait pertinent. Le but étant de process une date d'éligibilité, en fonction d'éléments qu'on passerait à ce modèle.

L'implémentation que je vois ça serait une factory ou côté publicodes, la fonction te renvoit en fonction d'élément que tu lui passes en paramètres (ex: je suis en CDD, ou CDI, etc.) l'éligibilité

Reply:
Je ne pense pas que l'on peut considérer le package "code-du-travail-modeles" comme étant le modele de notre mvc côté frontend (peut être qu'il faudrait changer le nom pour éviter la confusion). Le modèle du frontend c'est plutot le state de zustand.
Le package "code-du-travail-modeles" doit plutôt être vu comme l'équivalent d'une api qui renvoit un résultat calculé en fonction de paramètre d'entrée. Du coup, c'est assez logique de valider ces paramètres d'entrée avant de faire le calcul